js/index.js

```diff
@@ -1,42 +1,75 @@
+const newProductName = document.querySelector('.product-name input')
+const newProductPrice = document.querySelector('.product-price input')
 
 function updateSubtotal(product) {
-  console.log('Calculating subtotal, yey!');
-
-  //... your code goes here
+  const price = product.querySelector('.price span');
+  const quantity = product.querySelector('.quantity input')
+  const subTotal = product.querySelector('.subtotal span')
+  let totalValue = Number(price.innerHTML) * parseFloat(quantity.value)
+  subTotal.innerHTML = totalValue
+  return totalValue
 }
 
 function calculateAll() {
-  // code in the following two lines is added just for testing purposes.
-  // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
+  let endTotal = 0
+  const allProducts = document.querySelectorAll('.product')
 
-  // ITERATION 2
-  //... your code goes here
+  for (product of allProducts) {
+    endTotal += updateSubtotal(product)
+  }
 
-  // ITERATION 3
-  //... your code goes here
+  return displayTotal.innerHTML = endTotal
 }
 
-// ITERATION 4
-
 function removeProduct(event) {
   const target = event.currentTarget;
-  console.log('The target in remove is:', target);
-  //... your code goes here
+  itemList.removeChild(target.parentNode.parentNode)
+  calculateAll()
 }
 
-// ITERATION 5
+function activateRemoveButtons() {
+  const removeButtonsGroup = document.querySelectorAll('.btn-remove')
+  for (button of removeButtonsGroup) {
+    button.addEventListener('click', removeProduct)
+  }
+}
 
 function createProduct() {
-  //... your code goes here
+  let newProductRow = document.createElement('tr')
+
+  if (newProductName.value === '') {
+    alert(`Please enter a product name`)
+  } else if (newProductPrice.value === `0`) {
+    alert(`Please enter a price`)
+  } else {
+    newProductRow.innerHTML = `<td class="name">
+            <span>${newProductName.value}</span>
+          </td>
+          <td class="price">$<span>${newProductPrice.value}</span></td>
+          <td class="quantity">
+            <input type="number" value="0" min="0" placeholder="Quantity" />
+          </td>
+          <td class="subtotal">$<span>0</span></td>
+          <td class="action">
+            <button class="btn btn-remove">Remove</button>
+          </td>`
+    newProductRow.classList.add('product')
+    itemList.appendChild(newProductRow)
+    newProductName.value = ''
+    newProductPrice.value = '0'
+    activateRemoveButtons()
+  }
 }
 
 window.addEventListener('load', () => {
   const calculatePricesBtn = document.getElementById('calculate');
   calculatePricesBtn.addEventListener('click', calculateAll);
 
-  //... your code goes here
-});
+  activateRemoveButtons()
+
+  const createButton = document.querySelector('#create')
+  createButton.addEventListener('click', createProduct)
+});
```

Comment:
Great stuff Nina!
The UI works as expected.
I placed some comments regarding naming, style and const vs. let.